components/suggest/src/store.rs

```diff
@@ -344,6 +344,42 @@ where
                         // records, since we understand it now.
                         dao.drop_unparsable_record_id(&record_id)?;
 
+                        Ok(())
+                    })?;
+                }
+                SuggestRecord::Pocket => {
+                    let Some(attachment) = record.attachment.as_ref() else {
+                        // An Pocket record should always have an
+                        // attachment with suggestions. If it doesn't, it's
+                        // malformed, so skip to the next record.
+                        writer.write(|dao| dao.put_last_ingest_if_newer(record.last_modified))?;
+                        continue;
+                    };
+
+                    let attachment: SuggestAttachment<_> = serde_json::from_slice(
+                        &self.settings_client.get_attachment(&attachment.location)?,
+                    )?;
+
+                    writer.write(|dao| {
+                        // Drop any suggestions that we previously ingested from
+                        // this record's attachment. Suggestions don't have a
+                        // stable identifier, and determining which suggestions in
+                        // the attachment actually changed is more complicated than
+                        // dropping and re-ingesting all of them.
+                        dao.drop_suggestions(&record_id)?;
+
+                        // Ingest (or re-ingest) all suggestions in the
+                        // attachment.
+                        dao.insert_pocket_suggestions(&record_id, attachment.suggestions())?;
+
+                        // Remove this record's ID from the list of unparsable
+                        // records, since we understand it now.
+                        dao.drop_unparsable_record_id(&record_id)?;
+
+                        // Advance the last fetch time, so that we can resume
+                        // fetching after this record if we're interrupted.
+                        dao.put_last_ingest_if_newer(record.last_modified)?;
```

Comment:
Looks like this sequence of (drop-insert-drop-put) has been used for all the suggestion types, shall we wrap them up into another helper in DAO to keep `ingest_records()` DRY?